utils/utils.py

```diff
@@ -71,55 +72,188 @@ def train_validate_test_normal(
     scheduler,
     config,
     model_with_config_name,
+    plot_init_sol=True,
+    plot_hist_sol=False,
 ):
 
-    num_epoch = config["num_epoch"]
+    num_epoch = config["Training"]["num_epoch"]
+    trainlib = []
+    vallib = []
+    testlib = []  # total loss tracking for train/vali/test
+    tasklib = []
+    tasklib_test = []
+    tasklib_vali = []  # loss tracking for summation across all atoms/nodes
+    tasklib_nodes = []
+    tasklib_test_nodes = []
+    tasklib_vali_nodes = []  # probably not needed
+
+    if isinstance(model_wrapper, torch.nn.parallel.distributed.DistributedDataParallel):
```

Comment:
This could be a separate PR

Reply:
What could be?

Reply:
A `model_wrapper` function that returns model/module that we can call everywhere

Reply:
Yeah, I like the idea. Very neat. Let's do this in a separate PR.